### PR TITLE
feat: update gitignore to include package-lock.json in version control

### PR DESCRIPTION
**Pull request by greatislander**
Including a `package-lock.json` file in Git history is recommended. See here: https://nodejs.dev/learn/the-package-lock-json-file

> The `package-lock.json` file needs to be committed to your Git repository, so it can be fetched by other people, if the project is public or you have collaborators, or if you use Git as a source for deployments.

### CODE REVIEW

**Comment by jobara** [comment]
@greatislander fluid-project repos typically gitignore the package-lock.json file and pin to specific versions in the package.json file instead. This prevents issues of running unexpected versions between installs for example. @amb26 might have more to add about this.

**Comment by greatislander** [comment]
@jobara @amb26 I believe the same effect can be achieved by running `npm install` if a lock file is present— it installs the precise version recorded in the lock file.

Let's say I had started a project and added the `@11ty/eleventy-assets` dependency six months ago:

```bash
npm init --yes
npm install --save @11ty/eleventy-assets
```

The resulting `package.json` file would have had the following `dependencies`:

```js
  "dependencies": {
    "@11ty/eleventy-assets": "^1.0.1"
  }
```

However, the `package-lock.json` file pins the version to `1.0.1` precisely:

```js
  "dependencies": {
    ...
    "@11ty/eleventy-assets": {
      "version": "1.0.1",
      "resolved": "https://registry.npmjs.org/@11ty/eleventy-assets/-/eleventy-assets-1.0.1.tgz",
      "integrity": "sha512-sZcEo168nKwxTa4W+wdFq1qf7sctvr4M5fe1lzuoUAANwjTVcmArXaJe3I4UshrUgt51Q4WqiERgpWhm0Ai/0A==",
      "requires": {
        "dependency-graph": "^0.9.0"
      }
    },
```

The current version is 1.0.4, but running `npm install` with the `package-lock.json` file present still installs 1.0.1 even though the `package.json` file is not pinned to the originally installed version. The only way to install 1.0.4 with a lock file which specifies 1.0.1 is to run:

```bash
npm upgrade @11ty/eleventy-assets
```

It appears to me— and I could be wrong— that ignoring the lock file and manually pinning files avoids behaviour that the lock file exists to prevent. But I may be missing some context.

**Comment by amb26** [comment]
Hi all - it seems that at the time we made this recommendation there was a fault in the npm algorithm as described in these answers - https://stackoverflow.com/questions/45022048/why-does-npm-install-rewrite-package-lock-json - it would rewrite package.json on every npm install. As of npm 5.4.2 it seems that the fault is fixed.

**Comment by amb26** [comment]
> It seems if we did an npm install it would update the package-lock which would potentially bring in the updated sub dependencies
By my understanding of the answer and the linked response from the npm team https://github.com/npm/npm/issues/17979#issuecomment-332701215 it appears that npm install has now been fixed to never update a package-lock if it contains versions which are consistent with (even if different from) the ones specified in package.json. 
I agree that we should still pin versions in package.json, for which there is some discussion here: https://docs.renovatebot.com/dependency-pinning/

**Comment by greatislander** [comment]
It sounds like the conclusion here is that the `package-lock.json` file helps keep sub-dependencies pinned to a reproducible version, and the `package.json` file should include pinned versions for clarity. Is there somewhere we should update our guidance around this, say on the Wiki? If the reason we weren't versioning the lock file was because of that NPM bug there's no reason to keep doing so (and after reading that Renovate doc it sounds like keeping track of sub-dependency versions is a compelling reason to keep it in version control).

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,2 @@
 node_modules
-dist
-package-lock.json
+dist
```
Comment (jobara) [inline .gitignore:2]: seems to be missing a newline at the end of the file

**Comment by jobara** [comment]
@amb26 ran into another issue with a package-lock.json file while reviewing other PRs for me, not too long ago. @amb26 do you have any more thoughts on this, or are you okay for committing the package-lock.json file?

**Comment by amb26** [comment]
What was the issue you ran into?

**Comment by greatislander** [comment]
I got the sense @amb26 was happy with the change because he merged https://github.com/fluid-project/fluidic-11ty/pull/34, but I'd be curious to here what the other issues were.

**Comment by jobara** [comment]
> What was the issue you ran into?

It had to do with package-lock.json getting confused when testing the linting changes because I had to link to a GitHub endpoint instead of an NPM release. The package-lock.json saw these as the same version and didn't pull down the correct version. I believe that's what you ran into @amb26

**Comment by greatislander** [comment]
> It had to do with package-lock.json getting confused when testing the linting changes because I had to link to a GitHub endpoint instead of an NPM release. The package-lock.json saw these as the same version and didn't pull down the correct version.

So @jobara, if I'm understanding, when you had your branch of eslint-config-fluid installed via a Github reference here: https://github.com/fluid-project/infusion/pull/1013/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519

Running `npm install` would not pick up updates to your branch because the package-lock.json file had locked in a reference to a specific commit?

If so, I think that's expected behaviour and I would argue it's desirable. For testing a pull request, I think it would be best to reference a specific commit hash (the latest commit to the specific branch) in `package.json`, e.g.:

```diff
-        "eslint-config-fluid": "jobara/eslint-config-fluid#FLUID-6548",
+        "eslint-config-fluid": "jobara/eslint-config-fluid#a7c9060",
```

**Comment by amb26** [comment]
> It had to do with package-lock.json getting confused when testing the linting changes because I had to link to a GitHub endpoint instead of an NPM release. The package-lock.json saw these as the same version and didn't pull down the correct version. I believe that's what you ran into @amb26 

Yes, to me this is a recommendation against binding pulls against git revisions rather than npm dev versions rather than anything against committing package-lock.json itself. I'm not sure that @greatislander 's note accounts for all of the reasons npm may fail to properly update the tree although we should continue to perform more experiments. I've found that simply because npm considers that the package.json file that it has seen in one revision has a matching version to another is enough to confuse it in some situations.

**Comment by jobara** [comment]
Merged at 420fcb727ab6a08d58f15d50efa4be636af193e1

**Pull request merged**